File: 304.py

def myPow(base, exp):
    if exp == 0:
        return 1.0
    elif exp % 2 == 0:
        temp = myPow(base, exp // 2)
        return temp * temp
    else:
        temp = myPow(base, (exp - 1) // 2)
        return base * temp * temp


if __name__ == "__main__":
    base = float(input())
    exp = int(input())
    result = myPow(base, exp)
    print("{:.3f}".format(result))

File: 304.cpp
#include <iostream>
#include <iomanip> // For std::setprecision

using namespace std;

long double myPow(long double base, unsigned int exp) {
    if (exp == 0) {
        return 1.0;
    } else if (exp % 2 == 0) {
        long double temp = myPow(base, exp / 2);
        return temp * temp;
    } else {
        long double temp = myPow(base, (exp - 1) / 2);
        return base * temp * temp;
    }
}

int main() {
    long double base;
    unsigned int exp;
    cin >> base >> exp;
    
    long double result = myPow(base, exp);
    cout << fixed << setprecision(3) << result << endl;
    
    return 0;
}